ocs_ingester/fits.py

```diff
@@ -34,23 +34,49 @@ def __init__(self, file, required_headers, blacklist_headers):
             self.blacklist_headers.append('')
 
     def get_hdu_with_required_headers(self):
-        with self.file.get_fits() as fits_file:
-            with fits.open(io.BytesIO(fits_file.read()), mode='readonly') as hdulist:
-                for hdu in hdulist:
-                    fits_dict = dict(hdu.header)
-                    if any([k for k in self.required_headers if k not in fits_dict]):
-                        pass
-                    else:
-                        self.fits_dict = fits_dict
-                        logger.info('Ingester extracted fits headers', extra={
-                            'tags': {
-                                'request_num': fits_dict.get('REQNUM'),
-                                'PROPID': fits_dict.get('PROPID'),
-                                'filename': '{}{}'.format(self.file.basename, self.file.extension)
-                            }
-                        })
-                        return
-                raise DoNotRetryError('Could not find required headers!')  # No headers met requirement
+        if self.file.is_valid_fits():
+            # Loop through each HDU and use the first header that passes validation as the dict representation
+            with self.file.get_fits() as fits_file:
+                with fits.open(io.BytesIO(fits_file.read()), mode='readonly') as hdulist:
+                    for hdu in hdulist:
+                        fits_dict = dict(hdu.header)
+                        if self._is_valid_file_metadata(fits_dict):
+                            self.fits_dict = fits_dict
+                            return
+                        else:
+                            continue
+                    raise DoNotRetryError(
+                        'Could not find required keywords in headers!')  # No headers met requirement
+        # File provided is not a FITS file, use provided metadata
+        else:
+            if self.file.file_metadata is None:
+                raise DoNotRetryError('Metadata must be provided for non-FITS files.')
+            else:
+                if self._is_valid_file_metadata(self.file.file_metadata):
+                    self.fits_dict = self.file.file_metadata
+                else:
+                    raise DoNotRetryError(
+                        'Could not find required keywords in provided metadata!')  # No headers met requirement
+
+
+    def _is_valid_file_metadata(self, metadata_dict: dict):
+        """
+        Check some file metadata for required headers.
+
+        :param metadata_dict: dictionary of file metadata
+        :return True if required headers are present, False if not
+        """
+        if any([k for k in self.required_headers if k not in metadata_dict]):
+            return False
+        else:
+            logger.info('Ingester extracted fits headers', extra={
```

Comment:
I think this message could be misleading when it's being printed out when validating file metadata that was not extracted from a FITS file.... I mean, I guess it is technically still information that was in the associated FITS file header, but it wasn't extracted by the ingester.

Reply:
Good call. I think we can change this to "Ingester validated/extracted file metadata" or something like that, so it's more generalized.

Reply:
Updated to say: "Ingester validated file metadata," which is more in-line with what the code is _actually_ doing.